config/src/keys.rs

```diff
@@ -0,0 +1,233 @@
+        Self::parse(&contents)
+    }
+
+    fn parse(config_string: &str) -> Self {
+        toml::from_str(config_string).expect("Unable to parse Config")
+    }
+    // used to serialize keypairs to a configuration file
+    pub fn save_config<P: AsRef<Path>>(&self, output_file: P) {
+        let contents = toml::to_vec(&self).expect("Error serializing");
+        let mut file = File::create(output_file).expect("Error opening file");
+        file.write_all(&contents).expect("Error writing file");
+    }
+
+    // used in testing to fill the structure with test keypairs
+    pub fn load(
+        network_signing_private_key: Ed25519PrivateKey,
+        network_identity_private_key: X25519StaticPrivateKey,
+    ) -> Self {
+        let network_signing_public_key = (&network_signing_private_key).into();
+        let network_identity_public_key = (&network_identity_private_key).into();
+        Self {
+            network_signing_private_key: PrivateKeyContainer::Present(network_signing_private_key),
+            network_signing_public_key,
+            network_identity_private_key,
+            network_identity_public_key,
+        }
+    }
+
+    /// Beware, this destroys the private key from this NodeConfig
+    pub fn take_network_signing_private(&mut self) -> Option<Ed25519PrivateKey> {
+        self.network_signing_private_key.take()
+    }
+
+    pub fn get_network_identity_private(&self) -> X25519StaticPrivateKey {
+        self.network_identity_private_key.clone()
+    }
+
+    pub fn get_network_identity_public(&self) -> &X25519StaticPublicKey {
+        &self.network_identity_public_key
+    }
+
+    // getters for keypairs
+    pub fn get_network_identity_keypair(&self) -> (X25519StaticPrivateKey, X25519StaticPublicKey) {
+        (
+            self.get_network_identity_private(),
+            self.get_network_identity_public().clone(),
+        )
+    }
+}
+
+// ConsensusKeyPair is used to store a validator's consensus keypair.
+// It is filled via a config file at the moment.
+#[derive(Debug, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "testing"), derive(Clone))]
+pub struct ConsensusKeyPair {
+    consensus_private_key: PrivateKeyContainer<Ed25519PrivateKey>,
+    #[serde(serialize_with = "serialize_opt_key")]
+    #[serde(deserialize_with = "deserialize_opt_key")]
+    consensus_public_key: Option<Ed25519PublicKey>,
+}
+
+// required for serialization
+impl Default for ConsensusKeyPair {
+    fn default() -> Self {
+        let mut rng = StdRng::from_seed(TEST_SEED);
+        let (consensus_private_sig, consensus_public_sig) = compat::generate_keypair(&mut rng);
+        Self {
+            consensus_private_key: PrivateKeyContainer::Present(consensus_private_sig),
+            consensus_public_key: Some(consensus_public_sig),
+        }
+    }
+}
+
+impl ConsensusKeyPair {
+    // used to deserialize keypairs from a configuration file
+    pub fn load_config<P: AsRef<Path>>(path: P) -> Self {
```

Comment:
will we eventually return `Result<Self>` in these functions?

Reply:
yeah that would be ideal. For now, I'm unifying all `load_config` and `save_config` functions into a single trait which will make it easier.

Reply:
Will take that up later though